tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+        for second_point in [point, tuple(point), dict(zip(task._space.keys(), point))]:
+            second_results = task(second_point)
+            assert second_results[0]["name"] == first_results[0]["name"]
+            assert second_results[0]["type"] == first_results[0]["type"]
+            second_objective = second_results[0]["value"]
+
+            # TODO: Not sure why, but the two values are very close, but different!
+            assert np.isclose(first_objective, second_objective)
+
+    def test_call_with_gradients(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Test that calling the task with the `with_grad` returns the gradient at that point. """
+        task = self.Task(
+            max_trials=10,
+            task_id=0,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point_tuple = task._space.sample(1)[0]
+        point_dict = dict(zip(task._space.keys(), point_tuple))
+        results = task(point_dict, with_grad=True)
+
+        assert results[0]["type"] == "objective"
+        first_objective = results[0]["value"]
+        assert results[1]["type"] == "gradient"
+        first_gradient = results[1]["value"]
+
+        # TODO: Check if this testing logic makes sense, and the last condition isn't yet working.
+        step_size = 1.0
+
+        second_point_array = np.array(point_tuple) - step_size * np.array(first_gradient)
+
+        if any(isinstance(dim, _Discrete) for dim in task._space.values()):
+            # NOTE: Some dimensions don't quite work here because they are discrete: we can't really
+            # use a point that has batch_size = 12.04 for example.
+            assert second_point_array not in task._space
+
+        if second_point_array not in task._space:
+            with pytest.warns(
+                RuntimeWarning, match="isn't a valid point of the space"
+            ):
+                # NOTE: Bypassing the conversion that would normally happen in `dict_to_trial` for tuple
+                # inputs when a Tensor is passed directly to the task. When the point doesn't fit the
+                # space exactly, a RuntimeWarning is raised.
+                second_results = task(torch.as_tensor(second_point_array))
+        else:
+            # Check that no warnings are raised if the point is in the space of the task.
+            with pytest.warns(None) as record:
+                second_results = task(second_point_array)
+            assert len(record) == 0, [m.message for m in record.list]
+
+        second_objective = second_results[0]["value"]
+
+        # NOTE: (@lebrice): Expected this to work, but I might be a bit rusty.
+        # assert np.isclose(improvement, step_size)
```

Comment:
What is not working?

Reply:
The improvement isn't what I'd expect. Since I'm passing a value which is modified by + beta * (the gradients), I'd expect the output to improve by beta, right?

Reply:
Not necessarily, the gradient is local and you move by beta. It's true of the task is a linear function.

Reply:
Should work for a really really small step size though, no?

Reply:
Yep, but I would not call `1.0` a very small step size. :)